packages/@sanity/components/src/snackbar/SnackbarItem.js

```diff
@@ -0,0 +1,221 @@
+    onSetHeight: () => {},
+    setAutoFocus: false
+  }
+
+  DEFAULT_ICONS = {
+    info: <InfoIcon />,
+    success: <CheckCircleIcon />,
+    warning: <WarningIcon />,
+    error: <ErrorIcon />
+  }
+
+  snackIcon = () => {
+    const {icon, kind} = this.props
+    if (typeof icon === 'boolean' && icon) return this.DEFAULT_ICONS[kind]
+    if (typeof icon === 'object' || typeof icon === 'string') return icon
+    return undefined
+  }
+
+  handleAutoDismissSnack = () => {
+    const {autoDismissTimeout, isPersisted, id, onDismiss, onClose} = this.props
+    if (!isPersisted) {
+      this._dismissTimer = setTimeout(() => {
+        if (onClose) onClose()
+        onDismiss(id)
+      }, autoDismissTimeout)
+    }
+  }
+
+  handleMouseOver = () => {
+    this.cancelAutoDismissSnack()
+  }
+
+  handleMouseLeave = () => {
+    const {isPersisted} = this.props
+    if (!isPersisted) {
+      this.handleAutoDismissSnack()
+    }
+  }
+
+  handleAction = () => {
+    const {action, id, onDismiss} = this.props
+    if (action && action.callback) action.callback()
+    return onDismiss(id)
+  }
+
+  handleClose = () => {
+    const {id, onClose, onDismiss} = this.props
+    if (onClose) onClose()
+    return onDismiss(id)
+  }
+
+  cancelAutoDismissSnack = () => {
+    clearTimeout(this._dismissTimer)
+  }
+
+  componentDidMount() {
+    const {onSetHeight, id, isPersisted, setAutoFocus} = this.props
+
+    if (setAutoFocus) {
+      this._snackRef.current.focus()
+      this.cancelAutoDismissSnack()
+    }
+
+    const height = this._snackRef.current && this._snackRef.current.clientHeight
+    onSetHeight(id, height)
+
+    if (isPersisted) this.cancelAutoDismissSnack()
+    else this.handleAutoDismissSnack()
+
+    this._enterTimer = setTimeout(() => {
+      this.setState({
+        isEntering: false
+      })
+    }, 100)
+  }
+
+  componentWillUnmount() {
+    clearTimeout(this._dismissTimer)
+    clearTimeout(this._enterTimer)
+  }
+
+  render() {
+    const {
+      action,
+      children,
+      icon,
+      id,
+      isCloseable,
+      isOpen,
+      kind,
+      title,
+      subtitle,
+      offset
+    } = this.props
+
+    const rootStyles = this.state.isEntering
+      ? `${styles.root}`
+      : `${styles.root} ${isOpen ? styles.showSnack : styles.dismissSnack}`
+    const transition = `all 200ms ease-in-out`
+    const role = () => {
+      if (kind === 'success') return 'status'
+      if (kind === 'info') return 'log'
+      return 'alert'
+    }
+    return (
+      <div
+        aria-label={kind}
+        aria-describedby={`snackbarTitle-${kind}-${id}`}
+        role={role()}
+        ref={this._snackRef}
+        tabIndex="-1"
+        className={rootStyles}
+        style={{bottom: offset, transition: transition}}
+        onMouseOver={() => this.handleMouseOver()}
+        onMouseLeave={() => this.handleMouseLeave()}
+        onFocus={() => this.handleMouseOver()}
+        onBlur={() => this.handleMouseLeave()}
```

Comment:
nitpick, but the function name should be `handleBlur` and `handleFocus` here to have consistent variable names.  This is to avoid confusion when someone in the future when someone are looking only at `handleMouseover` and it gets triggered by something else.  Its a few more lines of code, but I think it is worth it.

Reply:
Done!